runbotics/runbotics-desktop/src/action/microsoft/automation/excel/cloud-excel.action-handler.ts

```diff
@@ -80,6 +82,36 @@ export class CloudExcelActionHandler extends StatefulActionHandler {
         return this.excelService.deleteColumns(this.session, columnRange);
     }
 
+    deleteRows(input: SharepointTypes.CloudExcelDeleteRowsActionInput) {
+        const { rowRange, worksheet } = input;
+
+        if (!rowRange.toString().match(ActionRegex.EXCEL_DELETE_ROWS_INPUT)) {
+            throw new Error(CloudExcelErrorMessage.deleteRowsIncorrectInput());
+        }
+
+        if (Array.isArray(rowRange)) {
+            const sortedDescendingRows = sortNumbersDescending(rowRange);
+            return this.deleteRowsOneByOne(this.session, sortedDescendingRows, worksheet);
+        } else if (rowRange.match(ActionRegex.EXCEL_ROW_RANGE)) {
+            return this.excelService.deleteRows(this.session, rowRange, worksheet);
+        } else if (rowRange.match(ActionRegex.EXCEL_DELETE_ROW_INPUT)) {
+            const address = `${rowRange}:${rowRange}`;
+            return this.excelService.deleteRows(this.session, address, worksheet);
+        } else {
+            let rows: number[];
+            try {
+                rows = JSON.parse(rowRange);
+                if (!Array.isArray(rows)) {
+                    throw new Error();
+                }
+            } catch (e) {
+                throw new Error(CloudExcelErrorMessage.deleteRowsIncorrectInput());
+            }
+            const sortedDescendingRows = sortNumbersDescending(rows);
+            return this.deleteRowsOneByOne(this.session, sortedDescendingRows, worksheet);
```

Comment:
I think in this block, if `try` fails, then `rows === undefined`, and then it is passed to `sortNumbersDescending`, which assumes you pass an array of strings or numbers. Imo, to prevent this, you can initialize `let rows: number[] = [];`